tests/integration/dynamic_ip_test.py

```diff
@@ -70,7 +70,10 @@
 IPV6_DEFAULT_GATEWAY = "::/0"
 IPV4_DEFAULT_GATEWAY = "0.0.0.0/0"
 
+LEASE_FILE = "/tmp/nmstate.dnsmasq.lease"
```

Comment:
Please use a `tempfile.NamedTemporaryFile` here.

Reply:
`NamedTemporaryFile()` require you to handle the file descriptor.

And based on https://stackoverflow.com/questions/26541416/generate-temporary-file-names-without-creating-actual-file-in-python , there is no good way to ask python generate a tmp file path without creating it and that file will be used by other user(here is dnsmasq).

As this is for test only where `/tmp/nmstate.dnsmasq.lease` can be assumed always accessable.

Reply:
Ok, if dnsmasq needs to create the file, please create a temporary directory with `tempfile.mkdtemp()` and make dnsmasq use that directory.

Reply:
The folder created by `tempfile.mkdtemp()` is only readable, writable, and searchable only by the creating user ID. Which does not allows dnsmasq to use it.

Reply:
You can use ´os.chown()` to give it to the dnsmasq user ID.

Reply:
@tyll , could you please explain why you insist on this point?
This is a test env, I do not see where this has issues.

Reply:
> @tyll , could you please explain why you insist on this point?
> This is a test env, I do not see where this has issues.

The failure happens when this is not running in a test environment (see the initial comment "When running DHCP integration test cases locally repeatly..."). In a test environment like a temporary container it would not be a problem because the lease would be removed anyhow.

Reply:
Here is the full python code to speed things up:

```python
import os
import pwd
import tempfile

def get_leasefile():
    dnsmasq_user = pwd.getpwnam("dnsmasq")
    tempdir = tempfile.mkdtemp(prefix="nmstate-dnsmasq-")
    os.chown(tempdir, dnsmasq_user.pw_uid, dnsmasq_user.pw_gid)
    return f"{tempdir}/dnsmasq.lease"
```

Reply:
The folder is removed at teardown and the only case that seems unsafe is that the same folder already existed. In a test env (whatever it is), it is up to the runner to make sure it will not interfere (i do not see how it can).

Reply:
There can be security implications if a privileged process creates files with predictable filenames in a world-writeable directory. Therefore it is bad practice to do so. There is an easy way to do it properly as I described. Other alternatives are to use `/var/lib/dnsmasq/dnsmasq-nmstate.leases` if you would like to use a predictable file name or to set the leasefile-ro option for dnsmasq to avoid using the lease database at all.

Reply:
The solution you requested is probably more correct but I still think the existing solution is good enough.
This is particular relevant when it is discussed for 4 days and for whatever reason the owner of the PR has not been convinced yet (or had no time to change it).

If the change is "good enough" and it can be easily perfected, you can just post an improvement after.

I think it is good enough because the folder name is living in the context of a test, and I do not see a security risk that it can present in practice. If you have something concrete that makes this a bad change, please convince us.

Reply:
IMHO you spend more effort on defending a bad solution than it takes to fix it. Therefore I added a commit to fix the issue. Hope this resolves it.

Reply:
I am trying to defend the project from standstill.
The proposal in the original PR is imperfect but not bad IMO.

I have no idea if your new commit solves the problem @cathay4t identify, so only he or someone who saw the problem can confirm.
I also think that it is inappropriate to inject your fix to the owner PR if it totally dismiss it . If you push it now, before the original change is merged, you can propose it through an alternative PR. (I did it several times and it was productive based on the outcome)

Reply:
This review is very unsatisfying. You create a lot of urgency to fix this issue but do not seem to be affected. Otherwise you could easily confirm my fix. So it only affects local testing for @cathay4t now and I guess therefore it is not a high priority for him right now. It does not affect the CI, since we use single-use containers there, so I do not see any justification for this urgency. You complain that it is too much effort to do this simple change, so I added it to keep it simple. But instead you want to make it even more effort by creating another PR while the simplest solution is to wait for @cathay4t so he can comment and adjust the PR as necessary.
I added my commit so he could easily used it as a fixup so we keep the discussion and fix together. However, since you did not like it, I removed it again. I am happy to push it back once we got a reaction from @cathay4t.